tests/lib/rules/no-extra-boolean-cast.js

```diff
@@ -31,7 +32,78 @@ ruleTester.run("no-extra-boolean-cast", rule, {
         "var foo = bar() ? Boolean(baz) : Boolean(bat)",
         "for(Boolean(foo);;) {}",
         "for(;; Boolean(foo)) {}",
-        "if (new Boolean(foo)) {}"
+        "if (new Boolean(foo)) {}",
+        {
+            code: "var foo = bar || !!baz",
+            options: [{ enforceForLogicalOperands: true }]
+        },
+        {
+            code: "var foo = bar && !!baz",
+            options: [{ enforceForLogicalOperands: true }]
+        },
+        {
+            code: "var foo = bar || (baz && !!bat)",
+            options: [{ enforceForLogicalOperands: true }]
+        },
+        {
+            code: "function foo() { return (!!bar || baz); }",
+            options: [{ enforceForLogicalOperands: true }]
+        },
+        {
+            code: "var foo = bar() ? (!!baz && bat) : (!!bat && qux)",
+            options: [{ enforceForLogicalOperands: true }]
+        },
+        {
+            code: "for(!!(foo && bar);;) {}",
+            options: [{ enforceForLogicalOperands: true }]
+        },
+        {
+            code: "for(;; !!(foo || bar)) {}",
+            options: [{ enforceForLogicalOperands: true }]
+        },
+        {
+            code: "var foo = Boolean(bar) || baz;",
+            options: [{ enforceForLogicalOperands: true }]
+        },
+        {
+            code: "var foo = bar || Boolean(baz);",
+            options: [{ enforceForLogicalOperands: true }]
+        },
+        {
+            code: "var foo = Boolean(bar) || Boolean(baz);",
+            options: [{ enforceForLogicalOperands: true }]
+        },
+        {
+            code: "function foo() { return (Boolean(bar) || baz); }",
+            options: [{ enforceForLogicalOperands: true }]
+        },
+        {
+            code: "var foo = bar() ? Boolean(baz) || bat : Boolean(bat)",
+            options: [{ enforceForLogicalOperands: true }]
+        },
+        {
+            code: "for(Boolean(foo) || bar;;) {}",
+            options: [{ enforceForLogicalOperands: true }]
+        },
+        {
+            code: "for(;; Boolean(foo) || bar) {}",
+            options: [{ enforceForLogicalOperands: true }]
+        },
+        {
+            code: "if (new Boolean(foo) || bar) {}",
+            options: [{ enforceForLogicalOperands: true }]
+        },
```

Comment:
These tests are great! Maybe we could also add one where there is something between the logical expression and the "boolean context", like this one:

```js
{
    code: "if ((!!foo || bar) === baz) {}",
    options: [{ enforceForLogicalOperands: true }]
}
```